atlasdb-client/src/main/java/com/palantir/atlasdb/ptobject/EncodingUtils.java

```diff
@@ -255,6 +257,25 @@ public static int sizeOfSizedBytes(byte[] bytes) {
         return Bytes.concat(len, bytes);
     }
 
+    public static byte[] encodeUUID(UUID uuid) {
+        byte[] arr = new byte[16];
+        System.arraycopy(Longs.toByteArray(uuid.getMostSignificantBits()), 0, arr, 0, 8);
+        System.arraycopy(Longs.toByteArray(uuid.getLeastSignificantBits()), 0, arr, 8, 8);
+        return arr;
+    }
+
+    public static UUID decodeUUID(byte[] bytes, int offset) {
+        long mostSigBits = PtBytes.toLong(bytes, offset);
+        long leastSigBits = PtBytes.toLong(bytes, offset + 8);
+        return new UUID(mostSigBits, leastSigBits);
+    }
+
+    public static UUID decodeFlippedUUID(byte[] bytes, int offset) {
+        long mostSigBits = PtBytes.toLong(flipAllBitsInPlace(bytes, offset, 8), offset);
```

Comment:
flip in place modifies the byte[].  I think just do PtBytes.toLong(bytes, offset) and PtBytes.toLong(bytes, offset + 8); and flip each with the standard ^ -1L trick.